s3-trigger-stack.yaml

```diff
@@ -79,9 +79,32 @@ Resources:
               response = {}
               try:
                   if event['RequestType'] == 'Delete':
-                      cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
-                      return {}
-                  
+                      try:
+
+                          event_data = event['ResourceProperties']
+                          lambda_arn = event_data.get('LambdaFunctionArn', '')
+                          buckets = event_data.get('S3BucketArns', [])
+                          for bucket_arn in buckets:
+                              parts = bucket_arn.split(":::")[-1].split("/", 1)
+                              bucket_name = parts[0]
+                           
+                              # get the existing notification configuration
+                              existing_config = s3.get_bucket_notification_configuration(Bucket=bucket_name)
+
+                              existing_lambda_configs = existing_config.get('LambdaFunctionConfigurations', [])
+                              updated_lambda_configs = [config for config in existing_lambda_configs if config['LambdaFunctionArn'] != lambda_arn]
```

Comment:
add comment to reflect that this is effectively deleting the lambda config.

Reply:
Added comment # Deleting the existing configuration